### ISSUE TITLE
Merge `twisted/python/_reflectpy3.py` into `twisted/python/reflect.py`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6239]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6239|
|Type|task|
|Created|2013-01-02 01:08:04Z|
|Branch|https://github.com/twisted/twisted/tree/merge-reflect-py3-6239|

Also merge the associated test module. See [#6183](https://github.com/twisted/twisted/issues/6183).

Attachments:

* [merge-reflect-py3-6239-1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/1a3/1a33779d57abe280e7d0e9e8f7b60f5c8488411e/20e46fe1c62537684117b6d70db4765727e98c22.patch) (75533 bytes) - added by multani on 2014-02-26 09:27:21Z - 
* [merge-reflect-py3-6239-2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/1a3/1a33779d57abe280e7d0e9e8f7b60f5c8488411e/7fb7d6b0bc78043decda2c527043b7fe3d88725c.patch) (76367 bytes) - added by multani on 2014-02-26 22:12:51Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6239 6239
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_merge_reflect_py3_6239 branches/merge-reflect-py3-6239
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1357088884000000 1357088884000000
changetime__1394974031000000 1394974031000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="multani's avatar" src="https://avatars.githubusercontent.com/u/65311?s=50" width="50" height="50">](https://github.com/multani)<a name="note_1"></a>|@multani commented|
|-|-|

This is also related to [#5929](https://github.com/twisted/twisted/issues/5929) (if not a dup).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra set status to `closed`|
|-|-|

Replying to [multani](#note_1):
> This is also related to [#5929](https://github.com/twisted/twisted/issues/5929) (if not a dup).

Marking it as a duplicate then.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

It isn't really a duplicate.  Merging the modules is different from porting the unported code in twisted/python/reflect.py.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `reopened`|
|-|-|

I was going to leave this closed because fixing it seemed like it would cause more trouble than it was worth.

However, then I noticed that there are 'two' patches attached to [#5929](https://github.com/twisted/twisted/issues/5929) - one of which is the proposed resolution to _this_ ticket.

So, in the interest of keeping this minimally confusing, I'm fixing this.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

See https://twistedmatrix.com/trac/attachment/ticket/5929/python3-reflect-5929-6.2.patch

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_6"></a>|@wallrj commented|
|-|-|

I don't think this can be reviewed until [#5929](https://github.com/twisted/twisted/issues/5929) is merged.

See ticket:5929#comment:34

**Comment by twisted-trac** [comment]
|[<img alt="multani's avatar" src="https://avatars.githubusercontent.com/u/65311?s=50" width="50" height="50">](https://github.com/multani)<a name="note_7"></a>|@multani set owner to @multani<br>@multani set status to `new`|
|-|-|

rwall is right, this ticket actually depends on [[#5929](https://github.com/twisted/twisted/issues/5929)](https://github.com/twisted/twisted/issues/5929).

I'll put it back in review with the appropriate patch attached when #5929 is done.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_8"></a>|@wallrj commented|
|-|-|

(In [41754]) Merge python3-reflect-5929-2

Author: multani
Reviewers: radix, thijs, rwall, glyph
Fixes: [#5929](https://github.com/twisted/twisted/issues/5929)
Refs: [#6239](https://github.com/twisted/twisted/issues/6239), [[#6986](https://github.com/twisted/twisted/issues/6986)](https://github.com/twisted/twisted/issues/6986), [[#6859](https://github.com/twisted/twisted/issues/6859)](https://github.com/twisted/twisted/issues/6859)

Port most of twisted.python.reflect to Python3.

Porting objgrep posed some problems and has been moved to a separate ticket #6986.

Porting getcurrent and isinst also posed problems and it was decided to deprecate them in  #6859.

**Comment by twisted-trac** [comment]
|[<img alt="multani's avatar" src="https://avatars.githubusercontent.com/u/65311?s=50" width="50" height="50">](https://github.com/multani)<a name="note_9"></a>|@multani commented|
|-|-|

This patch only:

 * moves the content of `twisted/python/_reflectpy3.py` into `twisted/python/reflect.py`
 * moves the content of `twisted/python/test/test_reflectpy3.py` into `twisted/test/test_reflect.py`
 * changes the import and references to the old module paths to the new ones

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

(In [41896]) Branching to merge-reflect-py3-6239.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_12"></a>|@hawkowl commented|
|-|-|

Looks good to me. Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_13"></a>|@hawkowl set status to `closed`|
|-|-|

(In [41904]) Merge merge-reflect-py3-6239: Merge `twisted/python/_reflectpy3.py` into `twisted/python/reflect.py`

Author: multani
Reviewer: hawkowl
Fixes: [#6239](https://github.com/twisted/twisted/issues/6239)

Merge `twisted/python/_reflectpy3.py` into `twisted/python/reflect.py`, along with the associated test cases.
